x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -214,19 +221,33 @@ private PhysicalOperation planAggregation(AggregateExec aggregate, LocalExecutio
         return physicalOperationProviders.groupingPhysicalOperation(aggregate, source, context);
     }
 
-    private PhysicalOperation planEsQueryNode(EsQueryExec esQuery, LocalExecutionPlannerContext context) {
-        if (esQuery.query() == null) {
```

Comment:
The query check and initialization happens inside physicalOperatorProviders method.